content/docs/other-guides/kustomize.md

```diff
@@ -0,0 +1,249 @@
+`kfctl init`.
+
+Below are some examples of `KfDef` configuration files:
+
+* [kfctl_k8s_istio.yaml](https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/kfctl_k8s_istio.yaml) 
+  to install Kubeflow on an existing Kubernetes cluster.
+* [kfctl_gcp_basic_auth.yaml](https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/kfctl_gcp_basic_auth.yaml) 
+  to set up a Google Kubernetes Engine (GKE) cluster with Kubeflow using basic
+  authentication.
+
+The kustomize package manager in kfctl uses the information in your
+`app.yaml` to traverse the directories under the 
+[Kubeflow manifests](https://github.com/kubeflow/manifests) and to 
+create kustomize build targets based on the manifests.
+
+## Installing kustomize
+
+Make sure that you have the minimum required version of kustomize:
+<b>{{% kustomize-min-version %}}</b> or later.
+
+1. Follow the [kustomize installation
+   guide](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/INSTALL.md),
+   choosing the relevant options for your operating system. For example, if
+   you're on Linux:
+
+    * Set some variables for the operating system:
+
+        ```
+        export opsys=linux
+        ```
+
+    * Download the kustomize binary:
+
+        ```
+        curl -s https://api.github.com/repos/kubernetes-sigs/kustomize/releases/latest |\
+        grep browser_download |\
+        grep $opsys |\
+        cut -d '"' -f 4 |\
+        xargs curl -O -L
+        ```
+
+    * Move the binary:
+
+        ```
+        mkdir -p ${HOME}/bin
+        mv kustomize_*_${opsys}_amd64 ${HOME}/bin/kustomize
+        chmod u+x ${HOME}/bin/kustomize
+        ```
+
+1. Include the `kustomize` command in your path:
+
+      ```
+      export PATH=$PATH:${HOME}/bin/kustomize
+      ```
+
+## Modifying configuration before deployment
+
+kustomize lets you customize raw, template-free YAML files for multiple
+purposes, leaving the original YAML untouched and usable as is.
+
+You can use the following command to build and apply kustomize directories:
+
+```
+kustomize build <kustomization_directory> | kubectl apply -f -
+```
+
+The [Kubeflow manifests repo](https://github.com/kubeflow/manifests) contains
+kustomize build targets, each with a `base` directory. You can use kustomize to
+generate YAML output and pass it to kfctl. You can also make
+changes to the kustomize targets in the manifests repo as needed. 
+
+## Modifying the configuration of an existing deployment
+
+To customize the Kubeflow resources running within the cluster, you can modify
+the kustomize manifests in `${KFAPP}/kustomize`, where `KFAPP` is the directory 
+where you stored your Kubeflow configurations during deployment. Then run
+`kfctl apply`.
+
+For example, to modify settings for the Jupyter web app within your Kubeflow
+deployment:
+
+1. Edit the configuration file at `${KFAPP}/kustomize/jupyter-web-app.yaml`.
```

Comment:
We should probably change these steps - it is better to change parameter values in the KfDef config file and regenerate.

Reply:
Ah I see - thanks! I'm working on an update and will commit it when ready.

Reply:
@richardsliu I'm having trouble figuring out which KfDef they should edit. Is it just the values in their `app.yaml` file? For example, my `app.yaml` has this for the Jupyter web app:

```
  - kustomizeConfig:
      overlays:
      - istio
      - application
      repoRef:
        name: manifests
        path: jupyter/jupyter-web-app
    name: jupyter-web-app
```

Reply:
Thanks!